File: 704_binary_search.rb
def search(nums, target)
  return -1 if !nums.include?(target)
  # Set low + mid + high?
  low = nums[0]
  high = nums[-1]
    mid = (low + high) / 2
  while target != mid do
    if mid < target
      low = mid
      mid = (low + high) / 2
    elsif mid > target
      high = mid
      mid = (low + high) / 2
    end
  end
  return nums.index(mid)
end


# nums = [-1,0,3,5,9,12]
nums = [2,5]
target = 5

p search(nums, target)

File: 704_binary_search.py
class Solution:
    def search(self, nums: list[int], target: int) -> int:
        # print(nums, target)
        min = 0
        max = len(nums) - 1
        while max >= min:
            middle = (min + max) // 2
            if target == nums[middle]:
                return middle
            elif target > nums[middle]:
                min = middle + 1
            else:
                max = middle - 1
        return -1

nums = [-1,0,3,5,9,12]
target = 9
# >>> 4

# nums = [-1,0,3,5,9,12]
# target = 2
# >>> -1

s = Solution()
print(s.search(nums, target))
